src/sage/modular/abvar/homspace.py

```diff
@@ -277,23 +277,19 @@ def _matrix_space(self):
         """
         return MatrixSpace(ZZ, 2*self.domain().dimension(), 2*self.codomain().dimension())
 
-    def _element_constructor_from_element_class(self, *args, **keywords):
-        """
-        Used in the coercion framework. Unfortunately, the default method
-        would get the order of parent and data different from what is expected
-        in ``MatrixMorphism.__init__``.
-
-        EXAMPLES::
+    def _coerce_map_from_(self, other):
+        r"""
+        TESTS::
 
-            sage: H = Hom(J0(11), J0(22))
-            sage: phi = H(matrix(ZZ,2,4,[5..12])); phi # indirect doctest
-            Abelian variety morphism:
-              From: Abelian variety J0(11) of dimension 1
-              To:   Abelian variety J0(22) of dimension 2
+            sage: t = J0(33).hecke_operator(2)
+            sage: t.parent().has_coerce_map_from(1)
```

Comment:
This test doesn't make sense (which is why it is failing).

Reply:
ah... thanks